src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -115,28 +115,59 @@ protected override void ApplyResponseChallenge()
         /// Responds to a 401 Challenge. Sends an OpenIdConnect message to the 'identity authority' to obtain an identity.
         /// </summary>
         /// <returns></returns>
+        /// <remarks>Uses log id's OIDCH-0026 - OIDCH-0050, next num: 37</remarks>
         protected override async Task ApplyResponseChallengeAsync()
         {
+            bool logLevelError = _logger.IsEnabled(LogLevel.Error);
+            bool logLevelDebug = _logger.IsEnabled(LogLevel.Debug);
+            bool logLevelInformational = _logger.IsEnabled(LogLevel.Information);
+            bool logLevelWarning = _logger.IsEnabled(LogLevel.Warning);
+
+            if (logLevelDebug)
+            {
+                if (logLevelDebug)
+                {
+                    _logger.LogDebug(string.Format(CultureInfo.CurrentCulture, Resources.OIDCH_0026_ApplyResponseChallengeAsync, this.GetType().ToString()));
+                }
+            }
+
             if (ShouldConvertChallengeToForbidden())
             {
+                if (logLevelDebug)
```

Comment:
@brentschmaltz the performance should be measured before making such a claim. We've already done work to ensure this is a very efficient operation. If you have evidence to the contrary, please file a bug in the https://github.com/aspnet/Logging/issues/ repo so we can address for everyone. In the meantime, please avoid the extra checks.